SmartDeviceLink/private/SDLProtocolReceivedMessageProcessor.m

```diff
@@ -0,0 +1,317 @@
+//
+//  SDLProtocolProcessMessageByte.m
+//  SmartDeviceLink
+//
+//  Created by George Miller on 7/13/22.
+//  Copyright © 2022 smartdevicelink. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "SDLProtocolHeader.h"
+#import "SDLProtocolReceivedMessageProcessor.h"
+#import "SDLProtocolReceivedMessageRouter.h"
+#import "SDLSecurityType.h"
+
+@implementation SDLProtocolReceivedMessageProcessor
+
+-(id)init {
+    self = [super init];
+    version = 0;
+    encrypted = false;
+    frameType = 0;
+    dataLength = 0;
+    dataBytesRemaining = 0; //Counter for the data pump
+    
+
+    // Message management
+    endOfMessage = 0;
+    header = nil;
+
+    //Reset state
+    [self ResetState];
+    return self;
+}
+
+- (void)ResetState{
+    // Flush Buffers
+    self.headerBuffer = [NSMutableData dataWithCapacity:0];
+    self.payloadBuffer = [NSMutableData dataWithCapacity:0];
+    dataBytesRemaining = 0;
+    
+    // Reset state
+    state = START_STATE;
+    prevState = ERROR_STATE;
+
+}
+
+// Loop through the given bytes and call the state machien to process each byte.
+- (void)stateMachineManager:(NSData *)receiveBuffer withBlock:(stateMachineMessageReadyBlock)messageReadyBlock{
+    //get a pointer to the bytes because NSMutableData is layered
+    const char *bytes = [receiveBuffer bytes];
```

Comment:
```suggestion
    const BytePtr bytes = (BytePtr)receiveBuffer.bytes;
```

1. Comment didn't really make sense.
2. I think this is equivalent code, but you should test.